client/state/data-layer/wpcom/sites/memberships/index.js

```diff
@@ -91,7 +91,7 @@ export const handleMembershipGetSettings = dispatchRequest( {
 		http(
 			{
 				method: 'GET',
-				path: `/sites/${ action.siteId }/memberships/status?source=${ action.source }`,
+				path: `/sites/${ action.siteId }/memberships/status?source=${ action.source ?? 'calypso' }`,
```

Comment:
I am not sure about that change...

Reply:
Defaults to `calypso` when `action.source` is `undefined` as `undefined` is not a valid `source`

Reply:
> Defaults to calypso when action.source is undefined as undefined is not a valid source

Thanks. My question was: when does this happens?

Reply:
I saw instances where `action.source` was `undefined` when trying to connect and disconnect Stripe from the Earn page

Reply:
Yes! Can you please be more precise and find those instance. That could unravel some other issue...

Reply:
1. Go to `https://wordpress.com/earn/payments/{site_slug}`
2. Disconnect Stripe
3. Inspect the Network panel
4. Notice that `undefined` is passed as a parameter for the `source` key


<img width="1724" alt="Screenshot 2023-07-07 at 12 22 39 PM" src="https://github.com/Automattic/wp-calypso/assets/7076981/34a447bd-a800-4e1e-889f-dcf2a02917b4">

<img width="1728" alt="Screenshot 2023-07-07 at 12 26 01 PM" src="https://github.com/Automattic/wp-calypso/assets/7076981/3d70f472-403a-40ab-ac39-283c6d863897">